### PR TITLE
Upgrade to sigma v5.0.5

### PR DESCRIPTION
**Pull request by aslesarenko**
In this PR:
- fixed ReducedErgoLikeTransaction serializer to store tx cost
- updated Sigma to v5.0.5 and code upgraded

### CODE REVIEW

File: common/src/main/java/org/ergoplatform/appkit/AppkitProvingInterpreter.scala [hunk 1]
```diff
@@ -202,32 +202,31 @@ class AppkitProvingInterpreter(
...
+        initCost = currentCost,
         activatedScriptVersion = (params.blockVersion - 1).toByte
       )
 
       val reducedInput = reduce(Interpreter.emptyEnv, inputBox.box.ergoTree, context)
 
-      currentCost = addCostLimited(currentCost,
-        reducedInput.reductionResult.cost, limit = maxCost, msg = inputBox.toString())
-
+      currentCost = reducedInput.reductionResult.cost
```
Comment (aslesarenko) [inline common/src/main/java/org/ergoplatform/appkit/AppkitProvingInterpreter.scala:212]: Note, reducedInput.reductionResult.cost includes context.initCost

File: common/src/main/java/org/ergoplatform/appkit/AppkitProvingInterpreter.scala [hunk 2]
```diff
@@ -244,15 +243,16 @@ class AppkitProvingInterpreter(
       val proverResult = proveReduced(reducedInput, unsignedTx.messageToSign)
       val signedInput = Input(unsignedInput.boxId, proverResult)
 
-      currentCost = addCostLimited(currentCost, proverResult.cost, maxCost, msg = signedInput.toString())
+      // TODO add verification cost here
```
Comment (MrStahlfelge) [inline common/src/main/java/org/ergoplatform/appkit/AppkitProvingInterpreter.scala:246]: not sure if it is good to add TODOs with this PR. maybe it should better wait until sigmastate has added the needed parts?

File: lib-impl/src/main/java/org/ergoplatform/appkit/impl/NodeAndExplorerDataSourceImpl.java [hunk 3]
```diff
@@ -66,7 +65,7 @@ public class NodeAndExplorerDataSourceImpl implements BlockchainDataSource {
      */
     public boolean performCheckBeforeSend = false;
 
-    public NodeAndExplorerDataSourceImpl(ApiClient nodeClient, @Nullable ExplorerApiClient explorerClient) {
+    public NodeAndExplorerDataSourceImpl(ApiClient nodeClient, ExplorerApiClient explorerClient) {
```
Comment (MrStahlfelge) [inline lib-impl/src/main/java/org/ergoplatform/appkit/impl/NodeAndExplorerDataSourceImpl.java:68]: the explorer is nullable

Comment (aslesarenko) [reply lib-impl/src/main/java/org/ergoplatform/appkit/impl/NodeAndExplorerDataSourceImpl.java:68]: Your reply in discord: "Yes you can remove it. The package is indeed weird normally it is from another namespace."

File: common/src/main/java/org/ergoplatform/appkit/AppkitProvingInterpreter.scala (hunk 2 above)
Comment (aslesarenko) [reply common/src/main/java/org/ergoplatform/appkit/AppkitProvingInterpreter.scala:246]: Implemented instead. This was a missing part of cost estimation.

File: common/src/main/java/org/ergoplatform/appkit/AppkitProvingInterpreter.scala [hunk 4]
```diff
@@ -162,7 +162,7 @@ class AppkitProvingInterpreter(
     // Cost of transaction initialization: we should read and parse all inputs and data inputs,
     // and also iterate through all outputs to check rules
     val initialCost = ArithUtils.addExact(
-      CostTable.interpreterInitCost,
+      10000, // use Interpreter.interpreterInitCost from Sigma once it's available
```
[the remark this answers is not in the document]
Comment (aslesarenko) [reply common/src/main/java/org/ergoplatform/appkit/AppkitProvingInterpreter.scala]: done

**Pull request merged**